slices/slices_test.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package slices
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestMap(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name   string
+		slice  []string
+		mapper func(string) int
+		want   []int
+	}{
+		{
+			name:   "nil_slice",
+			slice:  nil,
+			mapper: func(s string) int { return len(s) },
+			want:   nil,
+		},
+		{
+			name:   "empty_slice",
+			slice:  []string{},
+			mapper: func(s string) int { return len(s) },
+			want:   []int{},
+		},
+		{
+			name:   "one element",
+			slice:  []string{"foo"},
+			mapper: func(s string) int { return len(s) },
+			want:   []int{3},
+		},
+		{
+			name:   "many elements",
+			slice:  []string{"foo", "bar", "baz", "bing", "bang", "bong"},
+			mapper: func(s string) int { return len(s) },
+			want:   []int{3, 3, 3, 4, 4, 4},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
```

Comment:
As of Go 1.22, loop variable closures are not needed.

https://go.dev/blog/loopvar-preview

Reply:
this library is still on 1.21

Reply:
Brian is correct. Go 1.22 also isn't released yet. Once it's released, it will take 3-4 weeks for the ecosystem tooling to catch up. Then we can bump to 1.22 and consider dropping this. However, it will still work in 1.22 (it just isn't necessary), so we should continue doing this for now.